pil-analyzer/tests/types.rs

```diff
@@ -561,6 +561,61 @@ fn empty_conditional() {
     type_check(input, &[]);
 }
 
+#[test]
+fn defined_trait() {
+    let input = "
+    trait Add<T> {
+        add: T, T -> T,
+    }
+    impl Add<fe> {
+        add: |a, b| a + b,
+    }
+    let r: fe = Add::add(3, 4);
+    ";
+    type_check(input, &[("r", "", "fe")]);
+}
+
+#[test]
+#[should_panic = "Unable to derive concrete type for literal 2."]
```

Comment:
Since we don't support generic impls, could you move the test to the generic impl PR?